kvbc/src/categorization/kv_blockchain.cpp

```diff
@@ -453,6 +453,9 @@ void KeyValueBlockchain::deleteStateTransferBlock(const BlockId block_id) {
 void KeyValueBlockchain::deleteGenesisBlock() {
   // We assume there are blocks in the system.
   auto genesis_id = block_chain_.getGenesisBlockId();
+  // If a versioned/Merkle category contain more keys than concurrent_threshold
+  // It will be execute in a seperate thread.
+  const auto concurrent_threshold = 10;
```

Comment:
configuration?

Reply:
Same as above, I don't think there is a lot of room to tune it, so I decided to hard code it, but maybe it does worth the extra effort.